s3/pkg/datastore/yig/conf/yig.sql

```diff
@@ -43,19 +43,28 @@ DROP TABLE IF EXISTS `multiparts`;
 /*!40101 SET @saved_cs_client     = @@character_set_client */;
 /*!40101 SET character_set_client = utf8 */;
 CREATE TABLE `multiparts` (
-  `upload_id` bigint(20) UNSIGNED DEFAULT 0,
-  `part_num`  bigint(20)  DEFAULT 0,
-  `object_id` varchar(255) DEFAULT '',
-  `location` varchar(255) DEFAULT '',
-  `pool` varchar(255) DEFAULT '',
-  `offset` bigint(20) UNSIGNED DEFAULT 0, 
-  `size` bigint(20) UNSIGNED DEFAULT 0,
-  `etag` varchar(255) DEFAULT '',
-  `flag` tinyint(1) UNSIGNED DEFAULT 0,
-  `create_time` datetime DEFAULT CURRENT_TIMESTAMP,
-  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `upload_id` bigint(20) UNSIGNED NOT NULL DEFAULT 0,
+  `part_num`  bigint(20)  NOT NULL DEFAULT 0,
+  `object_id` varchar(255) NOT NULL DEFAULT '',
+  `location` varchar(255) NOT NULL DEFAULT '',
+  `pool` varchar(255) NOT NULL DEFAULT '',
+  `offset` bigint(20) UNSIGNED NOT NULL DEFAULT 0, 
+  `size` bigint(20) UNSIGNED NOT NULL DEFAULT 0,
+  `etag` varchar(255) NOT NULL DEFAULT '',
+  `flag` tinyint(1) UNSIGNED NOT NULL DEFAULT 0,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
   PRIMARY KEY (`upload_id`, `part_num`),
   UNIQUE KEY `rowkey` (`upload_id`,`part_num`,`object_id`, `flag`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;
 /*!40101 SET character_set_client = @saved_cs_client */;
 
+DROP TABLE IF EXISTS `gc` 
+CREATE TABLE `gc` (
+	`id` bigint(20) PRIMARY KEY AUTO_INCREMENT,
+	`location` varchar(255) NOT NULL DEFAULT '',
```

Comment:
What is the location?

Reply:
@kumarashit thanks for your review, the location is used by yig storage backend and it is the ceph cluster identifier.